services/github/github-deployments.service.js

```diff
@@ -0,0 +1,122 @@
+'use strict'
+
+const gql = require('graphql-tag')
+const Joi = require('@hapi/joi')
+const { renderBuildStatusBadge } = require('../build-status')
+const { GithubAuthV4Service } = require('./github-auth-service')
+const { documentation, transformErrors } = require('./github-helpers')
+
+const schema = Joi.object({
+  data: Joi.object({
+    repository: Joi.object({
+      deployments: Joi.object({
+        nodes: Joi.array()
+          .items(
+            Joi.object({
+              latestStatus: Joi.object({
+                state: Joi.string(),
+              }),
+            })
+          )
+          .required(),
+      }).required(),
+    }).required(),
+  }).required(),
+}).required()
+
+module.exports = class GithubDeployments extends GithubAuthV4Service {
+  static get category() {
+    return 'other'
+  }
+
+  static get route() {
+    return {
+      base: 'github/deployments',
+      pattern: ':user/:repo/:environment',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'GitHub deployments',
+        namedParams: {
+          user: 'badges',
+          repo: 'shields',
+          environment: 'shields-staging',
+        },
+        staticPreview: this.render({
+          state: 'success',
+        }),
+        documentation,
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'state' }
+  }
+
+  static render({ state }) {
+    const { label, message: build_message, color } = renderBuildStatusBadge({
+      label: 'state',
+      status: state,
+    })
+
+    let message = build_message
+
+    // renderBuildStatusBadge() does everything we need except it rewrites the
+    // message for status in 'greenStatuses' to 'passing'. This makes little
+    // sense in the context of a deployment. So rewrite it back to 'success'.
+    if (state == 'success') {
+      message = 'success'
+    }
```

Comment:
Given the above, is it really worth using the `renderBuildStatusBadge` helper here? Seems like all we'd be using it for here is to map a color to the `state` value from the deployment object